### PR TITLE
Bump OCP versions for sail operator

### PR DESCRIPTION
**Pull request by FilipB**

### CODE REVIEW

**Comment by FilipB** [comment]
/override ci/rehearse/periodic-ci-openshift-service-mesh-sail-operator-main-sync-upstream ci/rehearse/openshift-service-mesh/sail-operator/main/e2e-kind

**Comment by FilipB** [comment]
/pj-rehearse ci/rehearse/openshift-service-mesh/sail-operator/main/e2e-ocp-4-17-arm ci/rehearse/openshift-service-mesh/sail-operator/main/e2e-ocp-4-17 ci/rehearse/openshift-service-mesh/sail-operator/main/e2e-ocp-4-18 ci/rehearse/openshift-service-mesh/sail-operator/main/scorecard

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-scorecard  pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17-arm pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18 pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17-arm

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17-arm

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17-arm pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-17-arm pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

**Comment by FilipB** [comment]
/pj-rehearse pull-ci-openshift-service-mesh-sail-operator-main-e2e-ocp-4-18

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 base_images:
   cli:
-    name: "4.16"
+    name: "4.18"
```
[the remark this answers is not in the document]
Comment (FilipB) [reply ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:3]: I was trying both but better to use probably 4.17 which is GA already. 4.18 has only RC.1

[the remark this answers is not in the document]
Comment (yxun) [reply ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:3]: we don't need change here. The only place is related to the remote cluster version is using the env `HYPERSHIFT_HC_RELEASE_IMAGE` when provisioning a cluster using the hypershift workflow.

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml [hunk 2]
```diff
@@ -14,16 +14,17 @@ releases:
     release:
       architecture: arm64
       channel: stable
-      version: "4.16"
+      version: "4.17"
```
Comment (yxun) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:17]: we don't need change here. The only place is related to the remote cluster version is using the env `HYPERSHIFT_HC_RELEASE_IMAGE` when provisioning a cluster using the hypershift workflow.

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml [hunk 3]
```diff
@@ -162,15 +163,15 @@ tests:
...
+- as: e2e-ocp-4-18
   optional: true
   steps:
     cluster_profile: ossm-aws
     dependencies:
       OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:next
       RELEASE_IMAGE_LATEST: release:next
     env:
-      MAISTRA_BUILDER_IMAGE: quay-proxy.ci.openshift.org/openshift/ci:ci_maistra-builder_upstream-master
+      MAISTRA_BUILDER_IMAGE: gcr.io/istio-testing/build-tools:master-3d91e3d29bd1057b14995647d90cbf85e043eba6
```
Comment (yxun) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:174]:
```suggestion
      MAISTRA_BUILDER_IMAGE: gcr.io/istio-testing/build-tools:master-3d91e3d29bd1057b14995647d90cbf85e043eba6
       HYPERSHIFT_HC_RELEASE_IMAGE: quay.io/openshift-release-dev/ocp-release:4.18
      
```

I haven't seen a 4.18 tag from other hypershift usage in CI. We may need to wait for hypershift supporting a 4.18 builder image.

Comment (yxun) [reply ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:174]: we may try the `$RELEASE_IMAGE_LATEST`

File: ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml [hunk 4]
```diff
@@ -162,15 +163,15 @@ tests:
           memory: 200Mi
       timeout: 2h0m0s
     workflow: servicemesh-istio-e2e-hypershift
-- as: e2e-ocp-4-17
+- as: e2e-ocp-4-18
   optional: true
   steps:
     cluster_profile: ossm-aws
     dependencies:
       OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:next
```
Comment (yxun) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:171]:
```suggestion
      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
```

This tag is from openshift-cluster-control-plane-machine-set-operator-release-4.18_periodics.yaml
It should be a 4.18 version tag.

Comment (yxun) [inline ci-operator/config/openshift-service-mesh/sail-operator/openshift-service-mesh-sail-operator-main.yaml:171]: this `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` is a right way to use 4.18 builder image. I will add comment in the above section.

**Pull request merged**